cpp/order/CubaticOrderParameter.h

```diff
@@ -34,37 +70,95 @@ class CubaticOrderParameter
     void reset();
 
     //! Compute the cubatic order parameter
-    void compute(quat<float>* orientations, unsigned int n);
-
-    //! Calculate the cubatic tensor
-    void calcCubaticTensor(float* cubatic_tensor, quat<float> orientation);
-
-    void calcCubaticOrderParameter(float& cubatic_order_parameter, float* cubatic_tensor);
+    void compute(quat<float>* orientations, unsigned int num_orientations);
 
     //! Get a reference to the last computed cubatic order parameter
-    float getCubaticOrderParameter();
-
-    quat<float> calcRandomQuaternion(Saru& saru, float angle_multiplier);
+    float getCubaticOrderParameter()
+    {
+        return m_cubatic_order_parameter;
+    }
+
+    const util::ManagedArray<float> &getParticleOrderParameter()
+    {
+        return m_particle_order_parameter;
+    }
+
+    const util::ManagedArray<float> &getGlobalTensor()
+    {
+        return m_global_tensor;
+    }
+
+    const util::ManagedArray<float> &getCubaticTensor()
+    {
+        return m_cubatic_tensor;
+    }
+
+    unsigned int getNumParticles()
+    {
+        return m_n;
+    }
+
+    float getTInitial()
+    {
+        return m_t_initial;
+    }
+
+    float getTFinal()
+    {
+        return m_t_final;
+    }
+
+    float getScale()
+    {
+        return m_scale;
+    }
+
+    quat<float> getCubaticOrientation()
+    {
+        return m_cubatic_orientation;
+    }
+
+protected:
 
-    std::shared_ptr<float> getParticleCubaticOrderParameter();
-
-    std::shared_ptr<float> getParticleTensor();
-
-    std::shared_ptr<float> getGlobalTensor();
-
-    std::shared_ptr<float> getCubaticTensor();
-
-    std::shared_ptr<float> getGenR4Tensor();
-
-    unsigned int getNumParticles();
-
-    float getTInitial();
-
-    float getTFinal();
-
-    float getScale();
+    //! Calculate the cubatic tensor
+    /*! Implements the second line of eq. 27, the calculation of M_{\omega}.
+     *  The cubatic tensor is computed by creating the homogeneous tensor
+     *  corresponding to each basis vector rotated by the provided orientation
+     *  and then summing all these resulting tensors.
+     *
+     *  \param cubatic_tensor The cubatic tensor (denoted M_{\omega} in the paper), overwritten by reference.
+     *  \param orientation The orientation that will be used to determine the vectors used in the calculation.
+     */
+    tensor4 calcCubaticTensor(quat<float> &orientation);
+
+    //! Calculate the scalar cubatic order parameter.
+    /*! Implements eq. 22
+     *
+     *  \param cubatic_order_parameter The output value (updated as a reference)
```

Comment:
This comment is inaccurate. I'm not sure what to suggest.

Reply:
I think it's supposed to return `cubatic_order_parameter` and there's an argument `global_tensor` that's missing.